### ISSUE TITLE
Ignore index functionality

### ISSUE THREAD

**Issue by joshwilson-dev** [opengeos/geoai#765]
Description
I’d like to propose support for an ignore_index in the semantic segmentation training workflow, especially in geoai.train_segmentation_model() and SemanticSegmentationDataset.

At the moment, SemanticSegmentationDataset.__getitem__() normalizes label masks like this:

if self.num_classes == 2:
    if unique_vals.max() > 1:
        label_mask = (label_mask > 0).astype(np.int64)
else:
    label_mask = np.clip(label_mask, 0, self.num_classes - 1)

For multiclass segmentation, this means any ignored/no-data label such as PyTorch’s default -100 is clipped to class 0. Those pixels are then trained as background rather than ignored. This can significantly affect training when masks contain large no-data/ignore regions, which is common in geospatial segmentation workflows.

Expected outcome:

Add an ignore_index parameter to train_segmentation_model(), defaulting to -100 or None.
Pass it into SemanticSegmentationDataset.
Preserve ignored pixels during label normalization and resizing.
Use torch.nn.CrossEntropyLoss(ignore_index=ignore_index) when training.
Pass ignore_index into semantic segmentation metrics so validation metrics also exclude ignored pixels.
This would make GeoAI much safer for multiclass geospatial segmentation datasets where labels include no-data, masked-out areas, or unlabeled pixels.

Source code
Here is an example of the kind of change I’m proposing:

class SemanticSegmentationDataset(Dataset):
    def __init__(
        self,
        image_paths,
        label_paths,
        transforms=None,
        num_channels=None,
        target_size=None,
        resize_mode="resize",
        num_classes=2,
        ignore_index=-100,
    ):
        self.image_paths = image_paths
        self.label_paths = label_paths
        self.transforms = transforms
        self.target_size = target_size
        self.resize_mode = resize_mode
        self.num_classes = num_classes
        self.ignore_index = ignore_index
        ...
Then in __getitem__():

ignore_mask = None
if self.ignore_index is not None:
    ignore_mask = label_mask == self.ignore_index

unique_vals = np.unique(label_mask)

if self.num_classes == 2:
    if unique_vals.max() > 1:
        label_mask = (label_mask > 0).astype(np.int64)
else:
    label_mask = np.clip(label_mask, 0, self.num_classes - 1)

if ignore_mask is not None:
    label_mask[ignore_mask] = self.ignore_index
And in train_segmentation_model():

criterion = torch.nn.CrossEntropyLoss(ignore_index=ignore_index)

The resize path should also preserve ignored labels. For example, masks can still be resized with nearest-neighbor interpolation, but the result should not be clamped afterward in a way that converts ignore_index into a valid class.

I'm happy to add this feature myself if you'd like.
